sql/mozfun/norm/vpn_attribution/udf.sql

```diff
@@ -66,100 +67,58 @@ RETURNS STRUCT<
         -- normalized_acquisition_channel
         CASE
         WHEN
-          utm_source = "firefox-browser"
+          utm.source = "firefox-browser"
         THEN
           "Website: Firefox Browser"
         WHEN
-          utm_medium = '["referral","referral"]'
+          utm.medium = '["referral","referral"]'
           -- preserve legacy behavior
           AND (referrer LIKE "%app-store%") IS NOT TRUE
         THEN
           "Possible Attribution Error - Monitor"
         WHEN
           referrer LIKE "%app-store%"
-          AND utm_medium IS NULL
-          AND utm_source IS NULL
-          AND utm_campaign IS NULL
-          AND utm_content IS NULL
+          AND utm = ("(not set)", "(not set)", "(none)", "(direct)")
         THEN
           CONCAT("App Store: ", SPLIT(referrer, "-")[offset(2)])
         WHEN
           provider = "Apple Store IAP"
         THEN
           provider
         WHEN
-          normalized_medium IS NOT NULL
-          OR utm_source IS NOT NULL
-          OR utm_campaign IS NOT NULL
-          OR utm_content IS NOT NULL
+          normalized_medium != "(none)"
+          OR utm != ("(not set)", "(not set)", "(none)", "(direct)")
+          OR referrer IS NULL
         THEN
           "Website"
         ELSE
           "Unknown"
         END
         AS normalized_acquisition_channel,
-        -- normalized_campaign
-        IF(
-          utm_campaign IS NULL
-          AND (
-            (utm_content IS NULL AND utm_medium IS NULL AND utm_source IS NULL AND referrer IS NULL)
-            OR utm_source = "firefox-browser"
-            OR NULLIF(normalized_medium, '["referral","referral"]') IS NOT NULL
-          ),
-          "(not set)",
-          utm_campaign
-        ) AS normalized_campaign,
-        -- normalized_content
-        IF(
-          utm_content IS NULL
-          AND (
-            (
-              utm_campaign IS NULL
-              AND utm_medium IS NULL
-              AND utm_source IS NULL
-              AND referrer IS NULL
-            )
-            OR utm_source = "firefox-browser"
-            OR NULLIF(normalized_medium, '["referral","referral"]') IS NOT NULL
-          ),
-          "(not set)",
-          utm_content
-        ) AS normalized_content,
-        --
+        utm.campaign AS normalized_campaign,
+        utm.content AS normalized_content,
         normalized_medium,
         -- normalized_source
         CASE
         WHEN
-          utm_medium IS NULL
-          AND utm_source IS NULL
-          AND utm_content IS NULL
-          AND utm_campaign IS NULL
+          utm.source LIKE "mozilla.org-whatsnew%"
```

Comment:
The source might not have the www prefix before this point?

Reply:
source consistently lacks the www prefix and includes a version suffix for Fx <= 85